interfaces/apparmor/backend.go

```diff
@@ -294,6 +310,16 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 			logger.Noticef("cannot create host snap-confine apparmor configuration: %s", err)
 		}
 	}
+
+	// Deal with the "snapd" snap - we do the setup slightly differently
+	// here because this will run both on classic and on Ubuntu Core 18
+	// systems but /etc/apparmor.d is not writable on core18 systems
+	if snapName == "snapd" && release.AppArmorLevel() != release.NoAppArmor {
+		if err := setupSnapConfineReexec(snapInfo); err != nil {
```

Comment:
Perhaps we could pass a hint "snapd" to `setupSnapConfineReexec` to know where to look for, not sure if this helps.